### DISCUSSION TITLE
Azure DevOps source issue libgit2

### DISCUSSION THREAD

**Discussion by jsburckhardt** [fluxcd/flux2#2459] [Q&A] [answered]
Hi folks, I'm having a weird behaviour when trying to create a git source from an AzDO repo. We have a project with multiple repos, and I can create a source with all of them except for one. The one difference I can see between the repos is that the one that is failing is quite big ~130MB. 
I can confirm:
- I'm using libgit2
- running with the latest flux version
- the secret has rights to clone repo as I can use the secret to clone others
- I can clone the repo to my machine without any issues.
- Under /tmp it creates a folder with the name of the repo but then it stops

Here are some logs that didn't help a lot to troubleshoot.
```shell
◎ waiting for GitRepository source reconciliation
✗ unable to clone: User
```
After increasing the log-level to debug I can see an additional message but didn't really help me:
```JSON
{"level":"debug","ts":"2022-02-22T00:08:22.467Z","logger":"events","msg":"Normal","object":{"kind":"GitRepository","namespace":"flux-system","name":"reponame","uid":"bf956015-63ef-47cf-b3f5 │
│ -4e696fd5895e","apiVersion":"source.toolkit.fluxcd.io/v1beta1","resourceVersion":"1651745"},"reason":"error","message":"unable to clone: User"}                                          │
│ {"level":"error","ts":"2022-02-22T00:08:22.467Z","logger":"controller.gitrepository","msg":"Reconciler error","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitReposi │
│ tory","name":"remoname","namespace":"flux-system","error":"unable to clone: User","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\ts │
│ igs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/contr │
│ oller-runtime@v0.11.0/pkg/internal/controller/controller.go:227"}   
```

**Comment by stefanprodan** [comment, marked as answer]
Make sure your cluster is running the latest source-controller (upgrade to Flux v0.27.0) then increase the timeout in your GitRepository object to 5m and see if it clones it. Usually the libgit2 `User` error hides a timeout exception.

**Reply by jsburckhardt** [reply]
Cool, indeed, it was a timeout.
I increased the `--fetch-timeout` as someone suggested in the slack channel. That solved the issue ... kind of a pain libgit2 hides the error. Cheers

**Reply by stefanprodan** [reply]
We are going to surface the timeout errors in the next release.
